presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java

```diff
@@ -124,6 +127,7 @@ private HdfsContext(
         this.clientTags = Optional.of(session.getClientTags());
         this.tablePath = requireNonNull(tablePath, "tablePath is null");
         this.isNewTable = requireNonNull(isNewTable, "isNewTable is null");
+        this.metastoreHeaders = Optional.ofNullable(session.getProperty(METASTORE_HEADERS, String.class));
```

Comment:
call getMetastoreHeaders() in HiveSessionProperties

Reply:
This would be hard due to dependency ordering
HiveSessionProperties is in presto-hive
and this class is in presto-hive-common 
presto-hive depends on presto-hive-common, so if I add reverse dependency there will be cyclical dependency 
So I am not sure how can we avoid it?